packages/compass-aggregations/src/modules/pipeline-builder/pipeline-builder.ts

```diff
@@ -17,28 +17,63 @@ export const DEFAULT_PIPELINE = `[]`;
 const FULL_PIPELINE_PREVIEW_ID = Infinity;
 
 export class PipelineBuilder {
-  private _source: string = DEFAULT_PIPELINE;
+  source: string = DEFAULT_PIPELINE;
+  private _node: t.ArrayExpression | null = null;
   /* Pipeline representation of parsable source */
-  pipeline: Document[] | null = null;
-  node: t.ArrayExpression | null = null;
-  stages: Stage[] = [];
-  syntaxError: PipelineParserError[] = [];
+  private _pipeline: Document[] | null = null;
+  private _syntaxError: PipelineParserError[] = [];
+  private _stages: Stage[] = [];
   // todo: make private COMPASS-6167
   previewManager: PipelinePreviewManager;
 
   constructor(dataService: DataService, source = DEFAULT_PIPELINE) {
     this.previewManager = new PipelinePreviewManager(dataService);
-    this.source = source;
+    this.changeSource(source);
     this.sourceToStages();
   }
 
-  set source(source: string) {
-    this._source = source;
-    this.parseSourceToPipeline();
```

Comment:
Drive-by: the logic was spread between `changeSource` method and setter, so sometimes we would set the value and then call the `validateSource` method that changeSource calls, I aligned everything to use `changeSource` because it's how we do this for stages and what is being used in the redux slice